src/helpers/customElement.ts

```diff
@@ -0,0 +1,13 @@
+import { customElement as litCustomElement } from "lit/decorators.js";
+
+/**
+ * @description
+ * A call-through to the Lit `@customElement` decorator that also adds component
+ * information such as the tag name as a static property on the class.
+ */
+export function customElement(tagName: string) {
+  return (target: any) => {
+    litCustomElement(tagName)(target);
+    target["tagName"] = tagName;
```

Comment:
This is cool - but does it conflict with https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName ?

I can't tell what target is from the typing

Reply:
I also note you set tagName on the oe-subject-tag and and oe-task-meter classes - isn't that redundant now that you've done this?

Reply:
The add a typing patch to `AbstractComponent`?